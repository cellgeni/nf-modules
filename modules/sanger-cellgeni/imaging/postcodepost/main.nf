process IMAGING_POSTCODEPOST {
    tag "${meta.id}"
    label 'process_single'

    container "quay.io/cellgeni/postcode:0.2.1"

    input:
    tuple val(meta), path(model_params_and_losses_path), path(spot_loc_path), path(starfish_codebook_path)

    output:
    tuple val(meta), path("${decoded_profile}"), emit: decoded_profile
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    decoded_profile = "${prefix}_decoded.csv"
    """
    decode_postprocess.py run \\
        --model_file_path ${model_params_and_losses_path} \\
        --spot_locations_path ${spot_loc_path} \\
        --starfish_codebook_path ${starfish_codebook_path} \\
        -decoded_spots_df_path ${decoded_profile} \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        imaging: \$(decode_postprocess.py version)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    decoded_profile = "${prefix}_decoded.csv"
    """
    echo ${args}
    
    touch ${decoded_profile}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        imaging: \$(decode_postprocess.py version)
    END_VERSIONS
    """
}
