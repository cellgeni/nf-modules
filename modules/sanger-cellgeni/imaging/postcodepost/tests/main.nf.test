// nf-core modules test imaging/postcodepost
nextflow_process {

    name "Test Process IMAGING_POSTCODEPOST"
    script "../main.nf"
    process "IMAGING_POSTCODEPOST"

    tag "modules"
    tag "modules_sangercellgeni"
    tag "imaging"
    tag "imaging/postcodepost"

    test("predict - empty") {

        when {
            params.module_args = ''
            process {
                """
                input[0] = [
                    [ id:'merfish_1'],
                    file(params.modules_testdata_base_path + 'POSTCODE/merfish_1_model_params_and_losses.pt', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'MERFISH/for_decoding/merfish_1_starfish_codebook.json', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'MERFISH/merfish_transcript_locations_1.csv', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                // { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("predict - merfish_100000") {

        when {
            params.module_args = ''
            process {
                """
                input[0] = [
                    [ id:'merfish_100000'],
                    file(params.modules_testdata_base_path + 'POSTCODE/merfish_100000_model_params_and_losses.pt', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'MERFISH/for_decoding/merfish_100000_starfish_codebook.json', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'MERFISH/merfish_transcript_locations_100000.csv', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("predict - merfish_100000 - stub") {

        options "-stub"

        when {
            params.module_args = ''
            process {
                """
                input[0] = [
                    [ id:'stub' ], // meta map
                    file(params.modules_testdata_base_path + 'POSTCODE/merfish_100000_model_params_and_losses.pt', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'MERFISH/for_decoding/merfish_100000_starfish_codebook.json', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'MERFISH/merfish_transcript_locations_100000.csv', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
