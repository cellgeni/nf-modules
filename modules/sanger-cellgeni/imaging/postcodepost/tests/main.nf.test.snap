{
    "predict - merfish_100000": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "merfish_100000"
                        },
                        "merfish_100000_decoded.csv:md5,5fa67f527ac1baaca85079a657efdeeb"
                    ]
                ],
                "1": [
                    "versions.yml:md5,0bcd66ac7099d2c3498299b3488b9c6b"
                ],
                "decoded_profile": [
                    [
                        {
                            "id": "merfish_100000"
                        },
                        "merfish_100000_decoded.csv:md5,5fa67f527ac1baaca85079a657efdeeb"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0bcd66ac7099d2c3498299b3488b9c6b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-08-05T10:57:29.201099526"
    },
    "predict - merfish_5000": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "decoded_profile": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-08-05T10:50:19.264724704"
    },
    "predict - empty": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "decoded_profile": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-08-05T10:55:59.071477199"
    },
    "predict - merfish_100000 - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "stub"
                        },
                        "stub_decoded.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,0bcd66ac7099d2c3498299b3488b9c6b"
                ],
                "decoded_profile": [
                    [
                        {
                            "id": "stub"
                        },
                        "stub_decoded.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0bcd66ac7099d2c3498299b3488b9c6b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-08-05T09:55:03.922694633"
    },
    "predict - merfish_1": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "decoded_profile": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-08-05T10:50:10.306543847"
    }
}